src/mscorlib/src/System/Runtime/Loader/AssemblyLoadContext.cs

```diff
@@ -470,6 +470,27 @@ private void OnProcessExit(object sender, EventArgs e)
 
         // Synchronization primitive for controlling initialization of Default load context
         private static readonly object s_initLock = new Object();
+
+        // Occurs when an Assembly is loaded
+        public static event AssemblyLoadEventHandler AssemblyLoad 
+        { 
```

Comment:
Yes, it was response to your original question on why they are global.

On AppDomainAugments vs, just add them to ALC - it does not make much difference. They can be on ALC just fine.